framework/include/base/XFEMInterface.h

```diff
@@ -47,9 +48,11 @@ class XFEMInterface : public ConsoleStreamInterface
    * Constructor
    */
   explicit
-  XFEMInterface(MooseApp & app):
+  XFEMInterface(MooseApp & app, MooseSharedPointer<FEProblem> & fe_problem):
```

Comment:
Yes, however if smart pointers are expensive to copy then C++ has bigger
problems. Conceptually pointers should be light enough to be copied. How
many people do you think will get the nuances of a reference to a pointer?

On Wed, Apr 20, 2016 at 4:11 PM Daniel Schwen <notifications@github.com>
wrote:

> In framework/include/base/XFEMInterface.h
> <https://github.com/idaholab/moose/pull/6798#discussion_r60496769>:
>
> > @@ -47,9 +48,11 @@ class XFEMInterface : public ConsoleStreamInterface
> >     * Constructor
> >     */
> >    explicit
> > -  XFEMInterface(MooseApp & app):
> > +  XFEMInterface(MooseApp & app, MooseSharedPointer<FEProblem> & fe_problem):
>
> making it a const & might make the argument passing faster (no copy, no
> reference counting, and shared pointer is a non-trivial datastructure,
> right?)
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6798/files/0d77e5fcc87ab39a3ee3014818862edebad82406#r60496769>
>